rq/registry.py

```diff
@@ -268,6 +270,17 @@ def cleanup(self, timestamp: Optional[float] = None):
                         self.death_penalty_class, AbandonedJobError, AbandonedJobError(), traceback.extract_stack()
                     )
 
+                    # we did a job fetch so the meta is supposed to be there
+                    if job.meta:
+                        if ret := job.meta.get('_retried_after_abandonned', 0) < MAX_RETRIES_AFTER_ABANDONED:
+                            job.meta['_retried_after_abandonned'] = ret + 1
+                            job.retries_left = getattr(job, 'retries_left', 0) + 1
+                        else :
+                            logger.error(
+                                f'{self.__class__.__name__} failed to restart AbandonedJobError {job.id} '
```

Comment:
🌱  we didn't fail to restart, we explicitly stopped trying after X normal tries (might be 0) and Y abandoned_retries (I think this is helpful to put in the log)